UserDocs.md

```diff
@@ -14,245 +14,233 @@ You may wish create a Quilt-specific [profile](https://docs.aws.amazon.com/cli/l
+$ pip install git+https://github.com/quiltdata/t4.git#subdirectory=api/python
 ```
 
-## API reference
+## User guide
+### Creating a package
 
-> T4 expects S3 paths of the form `BUCKET_NAME/path/to/dir/data.csv`.
+To create a new in-memory package you initialize a new Package object and then perform operations on it until you have the Package you want. For example:
 
-### Reading and writing data
+	import t4
+	# initialize a package
+	p = t4.Package()
 
-![](img/helium-api.png)
+	# add entries individually using `set`
+	p = p.set("foo.csv", "/path/to/local/disk/foo.csv")
+	p = p.set("bar.csv", "s3://bucket/path/to/cloud/bar.csv")
 
+	# add many entries at once using `update`
+	p = p.update({"baz.csv": "/path/to/baz", "bam.png": "/path/to/bam"})
 
-#### `helium.get(src [, snapshot=None | version=None])`
-Retrieves `src` object from T4 and loads it into memory. Returns a `(data, metadata)` tuple.
+	# or grab everything in a directory at once using capture
+	p = p.capture("/path/to/folder/with/stuff/")
 
-Does not work on all objects. For a list of supported objects, see [Serialization](#serialization).
+	# delete entries using `delete`
+	p = p.delete("bam.png")
 
-Pass a snapshot hash to the optional `snapshot` parameter to retrieve the state of an S3 object at a given snapshot. Pass a version hash to the optional `version` parameter to retrieve the state of an S3 object at a given version. Only one or none of these two parameters may be specified at a time.
 
-#### `helium.put(src, dest, meta={})`
-Writes the in-memory object `src` to the path `dest` in S3.
-Transparently serializes some object types.
-See [Serialization](#serialization) for details.
+### Managing a local package
+A local package is one which has not been published yet. Local packages contain local data, so they’re highly malleable. The data they reference may change at any time.
 
-You can annotate your object with custom metadata by passing a `dict`
-to `meta=`. Object metadata are indexed and searchable by the helium API.
+If you would like to save or read a local package, you can use the following methods to do so.
 
-For more on metadata, see [Metadata and search](#metadata-and-search).
+	# save an empty package to the local registry
+	p = t4.Package()
+	tophash = p.build("example-package")
 
-#### `helium.get_file(src, dest [, snapshot=None | version=None])`
-Retrieves `src` object from T4, and writes to the `dest` file on your local disk.
+	# load the package from a local registry
+	p = t4.Package("example-package", hash=tophash)
 
-`snapshot=` and `version=` optional parameters, described in `helium.get()`.
+Packages are versioned by top hashes. Every time you build a package you are returned a top hash. To load a specific version of a package from a registry, as here, you can pass the top hash of the desired package version to the `hash` parameter. If you do not specify a top hash, the latest version will be retrieved.
```

Comment:
I would also add that you don't have to specify the package name if you specify the top hash